src/pyuvdata/uvbeam/uvbeam.py

```diff
@@ -4451,3 +4472,104 @@ def write_beamfits(self, filename, **kwargs):
         beamfits_obj = self._convert_to_filetype("beamfits")
         beamfits_obj.write_beamfits(filename, **kwargs)
         del beamfits_obj
+
+
+def uvbeam_constructor(loader, node):
+    """
+    Define a yaml constructor for UVBeam objects.
+
+    The yaml must specify a "filename" field pointing to the UVBeam readable file
+    and any desired arguments to the UVBeam.from_file method.
+
+    Parameters
+    ----------
+    loader: yaml.Loader
+        An instance of a yaml Loader object.
+    node: yaml.Node
+        A yaml node object.
+
+    Returns
+    -------
+    UVBeam
+        An instance of a UVBeam.
+
+    """
+    values = loader.construct_mapping(node)
+    if "filename" not in values:
+        raise ValueError("yaml entries for UVBeam must specify a filename.")
+
+    files_use = values["filename"]
+    if isinstance(values["filename"], str):
+        files_use = [values["filename"]]
+
+    if "path_variable" in values:
+        path_parts = (values.pop("path_variable")).split(".")
+        var_name = path_parts[-1]
+        if len(path_parts) == 1:
+            raise ValueError(
+                "If 'path_variable' is specified, it should take the form of a "
+                "module.variable_name where the variable name can be imported "
+                "from the module."
+            )
+        else:
+            module = (".").join(path_parts[:-1])
+            module = importlib.import_module(module)
+        path_var = getattr(module, var_name)
+        for f_i in range(len(files_use)):
+            files_use[f_i] = os.path.join(path_var, files_use[f_i])
```

Comment:
I'm not sure I understand the point of this? Is it that you have some UVBeams installed with pyuvdata that you'd like to be able to point to easily without knowing before hand where the installation directory is? I can imagine slicker interfaces to this, but perhaps this is the simplest way to achieve it.

Reply:
I could also imagine that, if the `filename` can be a list of files, that it might be useful to allow it to be a glob-style string.